File: ormliteimplementation/src/main/java/com/sayler/ormliteimplementation/OrmLiteManager.java
package com.sayler.ormliteimplementation;

import android.content.Context;
import com.annimon.stream.Stream;
import com.sayler.gina.domain.DataManager;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by miquido on 01/12/16.
 */

public class OrmLiteManager implements DataManager<List<BaseDataProvider>> {
  private List<BaseDataProvider> baseDataProviderList = new ArrayList<>();
  private Context context;

  public OrmLiteManager(Context context) {
    this.context = context;
  }

  public void add(BaseDataProvider baseDataProvider) {
    baseDataProviderList.add(baseDataProvider);
  }

  @Override
  public void setSourceFile(String sourceFilePath) {
    DBHelper.setDatabasePath(sourceFilePath);
    rebindProviders();
  }

  @Override
  public boolean isOpen() {
    return DBHelper.checkIfDatabaseFileExists();
  }

  @Override
  public void close() {
    //no used
  }

  @Override
  public List<BaseDataProvider> getDao() {
    return baseDataProviderList;
  }

  private void rebindProviders() {
    Stream.of(baseDataProviderList).forEach(dp -> dp.rebind(context));
  }
}

File: ormliteimplementation/src/main/java/com/sayler/ormliteimplementation/OrmLiteManager.kt
package com.sayler.ormliteimplementation

import android.content.Context
import com.annimon.stream.Stream
import com.sayler.gina.domain.DataManager

import java.util.ArrayList

/**

 */

internal class OrmLiteManager(private val context: Context) : DataManager<@JvmWildcard List<BaseDataProvider<*>>> {
    private lateinit var sourceFilePath: String
    private val baseDataProviderList = ArrayList<BaseDataProvider<*>>()

    fun add(baseDataProvider: BaseDataProvider<*>) {
        baseDataProviderList.add(baseDataProvider)
        rebindProviders()
    }

    override fun setSourceFile(sourceFilePath: String) {
        this.sourceFilePath = sourceFilePath
        DBHelper.setDatabasePath(sourceFilePath)
        rebindProviders()
    }

    override fun getSourceFilePath(): String {
        return sourceFilePath
    }

    override val isOpen: Boolean
        get() = DBHelper.checkIfDatabaseFileExists()

    override fun close() {
        //no used
    }

    override val dao: List<BaseDataProvider<*>>
        get() = baseDataProviderList

    private fun rebindProviders() {
        baseDataProviderList.forEach { it.rebind(context) }
    }
}
